payments-core/src/main/java/com/stripe/android/googlepaysheet/GooglePaySheetLauncher.kt

```diff
@@ -25,8 +49,13 @@ internal class DefaultGooglePaySheetLauncher(
 
     constructor(
         fragment: Fragment,
+        ioContext: CoroutineContext = Dispatchers.IO,
+        googlePayRepositoryFactory: (GooglePaySheetEnvironment) -> GooglePayRepository,
         callback: GooglePaySheetResultCallback
     ) : this(
+        fragment,
+        ioContext,
+        googlePayRepositoryFactory,
         fragment.registerForActivityResult(
```

Comment:
`registerForActivityResult` needs to be called every time the activity/fragment is created. Here we are the ones calling it as a result of `GooglePaySheet` being instantiated, which is not necessarily done every time the activity/fragment is created. So if the activity/fragment gets destroyed, it won't receive the result, which is a problem we're seeing frequently. I think we should start receiving the actual `ActivityResultLauncher` as a parameter across our API.

Reply:
The user must instantiate `GooglePaySheet` in `onCreate()`, which means that the Activity/Fragment will be registered to receive the result. Android guarantees that the result will be delivered even if the starting Activity is destroyed.

Reply:
This works if `GooglePaySheet` is always instantiated in `onCreate` or before, because when the caller is destroyed and recreated we will register for activity result again and the caller will receive the callback. If we want to keep doing things this way, which has its benefits like a simpler API that takes a callback object, then we need to make sure we're safeguarding against developers not instantiating `GooglePaySheet` when the activity/fragment is created (and instantiating it when a button is clicked, for example). Not sure if we can verify that, but we need to at least document that and log errors.

Reply:
Yeah, we definitely need to make it clear that registration needs to happen in `onCreate`. For what it's worth, this requirement also applies to `PaymentSheet`.

Reply:
If the user attempts to instantiate the object in `onResume`, they'll get a crash.

Reply:
What triggers the crash?

Reply:
Android will throw an exception if `registerForActivityResult` is called after a certain point in the lifecycle.

Reply:
It looks like we can enforce the state by verifying that `getLifecycle().getCurrentState()` is not CREATED, RESUMED or STARTED: https://developer.android.com/reference/androidx/lifecycle/Lifecycle.State. Not sure if there's any drawback to doing this

Reply:
Okay, so maybe that's what it does. So we only need to make sure we're always using `registerForActivityResult` and not calling `startActivityForResult`.

Reply:
👍